compiler/rustc_privacy/src/lib.rs

```diff
@@ -2020,10 +2121,20 @@ impl<'tcx> PrivateItemsInPublicInterfacesChecker<'tcx> {
                 if let hir::ItemKind::Impl(ref impl_) = item.kind {
                     let impl_vis =
                         ty::Visibility::of_impl(item.owner_id.def_id, tcx, &Default::default());
+                    let impl_ev = Some(
```

Comment:
Need a comment here explaining that we are using the non-shallow version here, unlike when building the eff vis table, because even if `Pub<Priv>` *can* leak things in general, it rarely happens in practice, and here we need a low number of false positives.